providers/google.go

```diff
@@ -262,7 +266,7 @@ func (p *GoogleProvider) RefreshSessionIfNeeded(ctx context.Context, s *sessions
 	}
 
 	// re-check that the user is in the proper google group(s)
-	if !p.ValidateGroup(s.Email) {
+	if !p.GroupValidator(s) {
```

Comment:
I'm not the biggest fan that this check happens again in the `RefreshSession` calls (where ideally we just want 1 quick network round trip since this is outside of the initial auth redeem/callback flow and in the middle of normal requests).  I would think providers should aim to avoid multiple network round trips in `RefreshSession`

But that leads to an interesting conundrum in providers that don't get Group details organically in the Access/ID Tokens and need a secondary call in `EnrichSessionState` to populate `session.Groups` where group membership may change (and a user shouldn't be authorized anymore).

I'm torn on the right call for this. Most secure is to get groups again on refresh.  Much more performant for providers like this is to only populate groups on initial session creation AND we advise users to not make a session too long if they are concerned about this scenario... 🤔